sys/linux/socket_netlink_route.txt

```diff
@@ -967,6 +970,113 @@ rtcanmsg {
+	LWTUNNEL_IP6_OPTS	nlattr[LWTUNNEL_IP6_OPTS, ip_opts_policy]
+
+# TODO: uncomment SEG6_IPTUNNEL_SRH, the related header above and the structs below,
+# after the kernel fixes the seg6_iptunnel.h uapi header.
+#	SEG6_IPTUNNEL_SRH	nlattr[SEG6_IPTUNNEL_SRH, tuninfo]
+#
+	LWT_BPF_IN		nlnest[LWT_BPF_IN, bpf_prog_policy]
+	LWT_BPF_OUT		nlnest[LWT_BPF_OUT, bpf_prog_policy]
+	LWT_BPF_XMIT		nlnest[LWT_BPF_XMIT, bpf_prog_policy]
+	LWT_BPF_XMIT_HEADROOM	nlattr[LWT_BPF_XMIT_HEADROOM, int32[0:LWT_BPF_MAX_HEADROOM]]
+
+	SEG6_LOCAL_ACTION	nlattr[SEG6_LOCAL_ACTION, int32[SEG6_LOCAL_ACTION_END:SEG6_LOCAL_ACTION_END_BPF]]
+	SEG6_LOCAL_SRH		nlattr[SEG6_LOCAL_SRH, ipv6_sr_hdr]
+	SEG6_LOCAL_TABLE	nlattr[SEG6_LOCAL_TABLE, int32]
+	SEG6_LOCAL_NH4		nlattr[SEG6_LOCAL_NH4, ipv4_addr]
+	SEG6_LOCAL_NH6		nlattr[SEG6_LOCAL_NH6, ipv6_addr]
+	SEG6_LOCAL_IIF		nlattr[SEG6_LOCAL_IIF, int32]
+	SEG6_LOCAL_OIF		nlattr[SEG6_LOCAL_OIF, int32]
+	SEG6_LOCAL_BPF		nlnest[SEG6_LOCAL_BPF, bpf_prog_policy_seg6_local]
+
+	RPL_IPTUNNEL_SRH	nlattr[RPL_IPTUNNEL_SRH, ipv6_rpl_sr_hdr]
+] [varlen]
+
+lwtunnel_ip_flags = TUNNEL_GENEVE_OPT, TUNNEL_VXLAN_OPT, TUNNEL_ERSPAN_OPT
+
+ip_opts_policy [
+	LWTUNNEL_IP_OPTS_GENEVE	nlnest[LWTUNNEL_IP_OPTS_GENEVE, geneve_opt_policy_lwt]
+	LWTUNNEL_IP_OPTS_VXLAN	nlnest[LWTUNNEL_IP_OPTS_VXLAN, vxlan_opt_policy_lwt]
+	LWTUNNEL_IP_OPTS_ERSPAN	nlnest[LWTUNNEL_IP_OPTS_ERSPAN, erspan_opt_policy_lwt]
+] [varlen]
+
+geneve_opt_policy_lwt [
+	LWTUNNEL_IP_OPT_GENEVE_CLASS	nlattr[LWTUNNEL_IP_OPT_GENEVE_CLASS, int16be]
+	LWTUNNEL_IP_OPT_GENEVE_TYPE	nlattr[LWTUNNEL_IP_OPT_GENEVE_TYPE, int8]
+	LWTUNNEL_IP_OPT_GENEVE_DATA	nlattr[LWTUNNEL_IP_OPT_GENEVE_DATA, array[int8, 0:128]]
+] [varlen]
+
+vxlan_opt_policy_lwt [
+	LWTUNNEL_IP_OPT_VXLAN_GBP	nlattr[LWTUNNEL_IP_OPT_VXLAN_GBP, int32]
+] [varlen]
+
+erspan_opt_policy_lwt [
+	LWTUNNEL_IP_OPT_ERSPAN_VER	nlattr[LWTUNNEL_IP_OPT_ERSPAN_VER, int8[1:2]]
+	LWTUNNEL_IP_OPT_ERSPAN_INDEX	nlattr[LWTUNNEL_IP_OPT_ERSPAN_INDEX, int32be]
+	LWTUNNEL_IP_OPT_ERSPAN_DIR	nlattr[LWTUNNEL_IP_OPT_ERSPAN_DIR, int8]
+	LWTUNNEL_IP_OPT_ERSPAN_HWID	nlattr[LWTUNNEL_IP_OPT_ERSPAN_HWID, int8]
+] [varlen]
+
+bpf_prog_policy [
+	LWT_BPF_PROG_FD		nlattr[LWT_BPF_PROG_FD, int32]
+	LWT_BPF_PROG_NAME	nlattr[LWT_BPF_PROG_NAME, string]
+] [varlen]
+
+bpf_prog_policy_seg6_local [
+	SEG6_LOCAL_BPF_PROG		nlattr[SEG6_LOCAL_BPF_PROG, int32]
+	SEG6_LOCAL_BPF_PROG_NAME	nlattr[SEG6_LOCAL_BPF_PROG_NAME, string]
+] [varlen]
+
+ipv6_rpl_sr_hdr {
+	nexthdr		int8
+	hdrlen		len[parent, int8]
+	type		const[RPL_IPTUNNEL_SRH, int8]
+	segments_left	int8
+	control		int32
+	address		array[ipv6_addr]
+} [packed]
+
+ila_attr_csum_mode = ILA_CSUM_ADJUST_TRANSPORT, ILA_CSUM_NEUTRAL_MAP, ILA_CSUM_NO_ACTION, ILA_CSUM_NEUTRAL_MAP_AUTO
+
+# TODO: describe remaining when supported
+ila_attr_ident_type = ILA_ATYPE_IID, ILA_ATYPE_LUID, ILA_ATYPE_USE_FORMAT
+
+ila_attr_hook_type = ILA_HOOK_ROUTE_INPUT, ILA_HOOK_ROUTE_OUTPUT
+
+#seg6_tuninfo {
```

Comment:
Is there any `TODO` for this? If it's just a leftover, please remove.

Reply:
There is a `TODO` in the same commit (sys/linux: add lwtunnel_encap_seg6 description), in the lwtunnel_policy union for SEG6_ENCAP.
Also, there is a an commented header for the same reason.
Should I write a `TODO` for each of them, because they aren't in the same place?

Reply:
If these are related, I would suggest adding a short TODO above them, so it's clearer. It's not immediately obvious these are related to the other TODO.

```
# TODO: uncomment after the kernel fixes the seg6_iptunnel.h uapi header.
```